hudi-client/hudi-client-common/src/main/java/org/apache/hudi/async/HoodieAsyncTableService.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.hudi.async;
+
+import org.apache.hudi.client.RunsTableService;
+import org.apache.hudi.config.HoodieWriteConfig;
+
+import java.util.function.Function;
+
+public abstract class HoodieAsyncTableService extends HoodieAsyncService implements RunsTableService {
```

Comment:
I see a reason to have RunsTableService, but whats the necessity to add HoodieAsyncService and then HoodieAsyncTableService. I feel we can drop HoodieAsyncService only. and HoodieAsyncTableService can directly implement RunsTableService.

Reply:
@nsivabalan i don't think we can drop `HoodieAsyncService` since the async services for clean, archive, compact and cluster all inherit from `HoodieAsyncService`. The `HoodieAsyncTableService` is to hold the common logic like this switch. Besides, when there is `HoodieAsyncService`, then it's a natural pattern to extend from it to make `HoodieAsyncTableService`. The opposite would look odd: why wouldn't `HoodieAsyncTableService` extend `HoodieAsyncService`

Reply:
@nsivabalan just recalled that `BaseHoodieWriteClient implements RunsTableService` as it needs to guard other sync APIs, so we'd need both `HoodieAsyncTableService` and `RunsTableService`